src/Core/src/Platform/iOS/TimePickerExtensions.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Globalization;
+using System.Linq;
+using Foundation;
+using UIKit;
+
+namespace Microsoft.Maui
+{
+	public static class TimePickerExtensions
+	{
+		public static void UpdateFormat(this MauiTimePicker nativeView, ITimePicker view)
+		{
+			nativeView.UpdateTime(view, null);
+		}
+
+		public static void UpdateFormat(this MauiTimePicker nativeView, ITimePicker view, UIDatePicker? picker)
+		{
+			nativeView.UpdateTime(view, picker);
+		}
+
+		public static void UpdateTime(this MauiTimePicker nativeView, ITimePicker view)
+		{
+			nativeView.UpdateTime(view, null);
+		}
+
+		public static void UpdateTime(this MauiTimePicker nativeTimePicker, ITimePicker timePicker, UIDatePicker? picker)
+		{
+			if (picker != null)
+				picker.Date = new DateTime(1, 1, 1).Add(timePicker.Time).ToNSDate();
+
+			string iOSLocale = NSLocale.CurrentLocale.CountryCode;
+			var cultureInfo = CultureInfo.GetCultures(CultureTypes.AllCultures)
+				.Where(c => c.Name.EndsWith("-" + iOSLocale)).FirstOrDefault();
+
+			if (cultureInfo == null)
+				cultureInfo = CultureInfo.InvariantCulture;
+
+			if (string.IsNullOrEmpty(timePicker.Format))
+			{
+				string timeformat = cultureInfo.DateTimeFormat.ShortTimePattern;
+				NSLocale locale = new NSLocale(cultureInfo.TwoLetterISOLanguageName);
+
+				if (picker != null)
+					picker.Locale = locale;
+			}
```

Comment:
I see a lot of this in the TimeExtensions.

Reply:
A lot of what?

Reply:
Most of this logic. Is it duplicated and you meant to use that instead? https://github.com/dotnet/maui/pull/454/files#diff-edd784be3a5dd0cdfa4ef9f7391413297e0c202a70445e81e4e5163e4b411ed6R20-R25

Reply:
Maybe what I mean is that this could be a separate method?

Reply:
Oh, yeah, some of that looks like it could probably be de-duplicated.